### ISSUE TITLE
Let holes behave like `case {}` wrt multiplicities

### ISSUE THREAD

**Issue by aspiwack** [tweag/ghc#268]
The short story is that whenever you type something like

```haskell
f :: A ->. B ->. C
f x y = _
```

You get a bunch of multiplicity errors. Which are irrelevant to you and distracting.

But that's ok! We can do better! We can make it so that `_` consumes `x` and `y` with arbitrary multiplicity, just like `case {}` does. That is, just like `case {}` will when #25 is resolved.

I believe that to be the expected behaviour.

**Comment by monoidal** [comment]
We could even show the variables with their expected multiplicities in the context.

**Comment by aspiwack** [comment]
It's orthogonal, but indeed. Probably all non-omega bindings in error messages etc… should come with their multiplicity.

**Comment by buggymcbugfix** [comment]
Sorry to chime in like this, but consider 
```haskell
f :: A ->. B ->. (A, B)
f x y = (_,_)
```

I talked to Edwin about this in relation to Idris 2. The behaviour was that the context of the rhs hole would not show `x` and `y` as available because it assumed that you consume all of them in the first hole since you can't consume them in both holes.

However, you may very well want to fill in the right hand side hole first, so we decided that it would be best to change that behaviour in order to show variables with their multiplicities in all holes. If you do end up using it in both holes you will obviously get a linearity error next time you run the type checker, but it's a good idea to work on one hole at a time anyway.

**Comment by aspiwack** [comment]
Not at all. Thoughts always, and very, welcome.

I don't think that the conclusion is quite correct: `case {}` doesn't eat every variables it sees, preventing the rest of the expression to see it:

```haskell
f :: A ->. B ->. (A, B) f x y = (case {},y)
```

Is a well-typed expression. So I think “like `case {}`” is still the right behaviour (it's not super well defined, but it's ok).

Besides, in the context-and-usage presentation, even variables which have been fully used up elsewhere appear in the context. Quite possibly without a multiplicity at all (if there were a multiplicity, it would be the one on the binder, not adjusted for the rest of the expression: the algorithmic presentation doesn't do subtractions).

Is Idris 2 any different in that respect?

**Comment by buggymcbugfix** [comment]
Thanks for giving the example—that makes sense.

Last time Edwin showed me, with holes Idris 2 will give the context together with the available quantities after subtracting any usages, e.g. for the example I gave it would give the context
```
1 A
1 B
```
whereas for your example it would give
```
1 A
0 B
```

I'm planning to have a closer look at Idris 2 next week with a group of people and if things have changed (or I misremembered something), I will let you know.
